sklego/linear.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import Union
+
+import numpy as np
+import scipy.spatial.distance as distance
+from sklearn.base import BaseEstimator, RegressorMixin
+from sklearn.linear_model import LinearRegression
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils import check_X_y
+from sklearn.utils import check_array
+from sklearn.utils.validation import FLOAT_DTYPES
+from sklearn.utils.validation import check_is_fitted
+
+
+class LoessRegressor(BaseEstimator, RegressorMixin):
+    """
+    Lo(w)ess regressor.
+
+    """
+
+    def __init__(self, weighting_method: Union[str, None] = None, span: float = .1):
+        super().__init__()
+
+        self._check_init_inputs(weighting_method, span)
+
+        self.weighting_method = weighting_method
+        self.span = span
+        self.xs = None
+        self.ys = None
+        self.dim_ = None
+        self.logger = logging.getLogger(__name__)
+
+    def fit(self, X: np.array, y: np.array):
+        """
+        Fit the regressor on the data by storing the data sorted on the inputs.
+
+        :param X: Array-like object of shape (n_samples, 1), input data for the model.
+        :param y: Array-like object of shape (n_samples, 1), target output data for the model
+        :return: The fitted instance
+        """
+
+        self.xs, self.ys = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+        self.dim_ = self.xs.shape[1]
+
+        return self
+
+    def predict(self, X: np.array, with_indices: bool = False) -> Union[np.array,
+                                                                        (np.array, np.array)]:
+        """
+        Predict targets using the fitted model
+        :param X: Array-like object of shape (n_samples, 1), input data for the model.
+        :param with_indices: If set to True, function also returns the indices of the sub-set in the
+            train data used for fitting the local model.
+        :return: Array-like object of shape (n_samples, 1), predicted target output. If with_indices
+            is set to true, a tuple of two array-like objects (y_predictions, indices) is returned
+            instead.
+        """
+
+        check_is_fitted(self, ['dim_'])
+        X = check_array(X)
+
+        y_pred = np.array([])
+
+        if with_indices:
+            indices = []
+
+        for x in X:
+            idx_window = self._get_window_indices(x.reshape(-1, 1))
+
+            if with_indices:
+                indices.append(idx_window)
+
+            X = self.xs[idx_window].reshape(-1, 1)
+            y = self.ys[idx_window]
+
+            weights = self._create_weights(x, X)
+            model = LinearRegression().fit(X, y, sample_weight=weights)
+
+            y_pred = np.append(y_pred, model.predict(x.reshape(-1, 1)))
```

Comment:
it is more performant not to append. instead it might be better to initialise an array with zeros and then fill the values by assignment.

Reply:
Check